src/qutip_qip/circuit/text_renderer.py

```diff
@@ -0,0 +1,497 @@
+        Draw a multi qubit gate
+
+        Parameters
+        ----------
+        gate : Gate
+            The gate to be drawn.
+
+        Returns
+        -------
+        tuple
+            The parts of the gate to be drawn.
+            i.e. the top_frame, mid_frame, mid_connect, mid_connect_label, and bot_frame.
+
+        int
+            The width of the gate.
+        """
+
+        lid_seg = "─" * (ceil(self.style.gate_pad) * 2 + len(gate_text))
+        pad = " " * ceil(self.style.gate_pad)
+
+        top_frame = f" ┌{lid_seg}┐ "
+        bot_frame = f" └{lid_seg}┘ "
+        mid_frame = f" │{pad}{' ' * len(gate_text)}{pad}│ "
+        mid_connect = f"─┤{pad}{' ' * len(gate_text)}{pad}├─"
+        mid_connect_label = f"─┤{pad}{gate_text}{pad}├─"
+
+        # Adjust top_frame or bottom if there is a control wire
+        if gate.controls:
+            mid_index = len(bot_frame) // 2
+            if gate.controls[0] < gate.targets[0]:
+                bot_frame = (
+                    bot_frame[:mid_index] + "┬" + bot_frame[mid_index + 1 :]
+                )
+            else:
+                top_frame = (
+                    top_frame[:mid_index] + "┴" + top_frame[mid_index + 1 :]
+                )
+
+        return (
+            top_frame,
+            mid_frame,
+            mid_connect,
+            mid_connect_label,
+            bot_frame,
+        ), len(top_frame)
+
+    def _draw_measurement_gate(self, measurement: Measurement):
+        """
+        Draw a measurement gate
+        """
+
+        parts, width = self._draw_singleq_gate("M")
+        top_frame, mid_frame, bot_frame = parts
+
+        # adjust top_frame or bottom according the placement of the classical wire
+        mid_index = len(bot_frame) // 2
+        if measurement.classical_store + self._qwires > measurement.targets[0]:
+            bot_frame = (
+                bot_frame[:mid_index] + "╥" + bot_frame[mid_index + 1 :]
+            )
+        else:
+            top_frame = (
+                top_frame[:mid_index] + "╨" + top_frame[mid_index + 1 :]
+            )
+
+        return (top_frame, mid_frame, bot_frame), width
+
+    def _update_cbridge(
+        self, gate: Gate, wire_list: List[int], xskip: int, width: int
+    ):
+        """
+        Update the render strings for the control bridge
+
+        Parameters
+        ----------
+        wire_list : list
+            The list of control wires the gate is acting on and all the wires in between.
+
+        xskip : float
+            The horizontal value for getting to requested layer.
+
+        width : int
+            The width of the gate.
+        """
+
+        bar_conn = " " * (width // 2) + "║" + " " * (width // 2)
+        mid_bar_conn = "─" * (width // 2) + "║" + "─" * (width // 2)
+        mid_bar_classical_conn = "═" * (width // 2) + "║" + "═" * (width // 2)
+        classical_conn = "═" * (width // 2) + "╩" + "═" * (width // 2)
+        self._adjust_layer(wire_list, xskip)
```

Comment:
There is an `_adjust_layer` for all `_update_*` methods except for `_update_singleq`, why?

Reply:
Padding via `_adjust_layer` is primarily necessary when a gate interacts with multiple wires, as it ensures consistent padding across all of them.

However, while testing with align_layer=True, I discovered an edge case where additional padding is needed. I'll likely implement this adjustment after each gate is added.